### ISSUE TITLE
`missing_panics_doc` shouldn't be triggered in constant environments

### ISSUE THREAD

**Issue by c410-f3r** [rust-lang/rust-clippy#12760]
IFAICT, documentation around `panic!` is about runtime behavior. `rustc` will automatically point to the panic message at compile-time when working with constants.

```rust
#![deny(clippy::missing_panics_doc)]

pub fn foo<const N: usize>() {
    const {
        if N == 0 {
            panic!();
        }
    }
}

pub fn main() {
    foo::<0>();
}
```
